Validator/src/main/java/org/openzen/zenscript/validator/visitors/DefinitionMemberValidator.java

```diff
@@ -119,22 +121,36 @@ public Void visitConstructor(ConstructorMember member) {
                 .getMemberCache());
 		
 		if (member.body == null) {
+			if(member.getBuiltin() == BuiltinID.CLASS_DEFAULT_CONSTRUCTOR) {
+				checkConstructorForwarded(member);
+			}
+
 			if (!member.isExtern()) {
 				validator.logError(ValidationLogEntry.Code.BODY_REQUIRED, member.position, "Constructors must have a body");
 				return null;
 			}
 		} else {
-			StatementValidator statementValidator = new StatementValidator(validator, new ConstructorStatementScope(member.header, member.getAccessScope()));
-			member.body.accept(statementValidator);
-			validateThrow(member, member.header, member.body);
-			
-			if (member.definition.getSuperType() != null && !statementValidator.constructorForwarded) {
-				validator.logError(ValidationLogEntry.Code.CONSTRUCTOR_FORWARD_MISSING, member.position, "Constructor not forwarded to base type");
-			}
+			checkConstructorForwarded(member);
 		}
 		
 		return null;
 	}
+
+	private void checkConstructorForwarded(ConstructorMember member) {
+		final Statement body = member.body == null ? new EmptyStatement(member.position) : member.body;
+		StatementValidator statementValidator = new StatementValidator(validator, new ConstructorStatementScope(member.header, member.getAccessScope()));
+		body.accept(statementValidator);
+		validateThrow(member, member.header, body);
+
+		if (member.definition.getSuperType() != null && !statementValidator.constructorForwarded) {
+			//TODO: If the parent has an empty constructor, can we allow it?
+			if(member.definition.getSuperType() instanceof DefinitionTypeID && ((DefinitionTypeID) member.definition.getSuperType()).definition.hasEmptyConstructor()) {
```

Comment:
@stanhebben I was unsure on this one, is it okay not to log an error in this case?
This allows constructors to not have a `this` or `super` call if the super type has a no-arg constructor

Reply:
Yeah that's fine